node/popn.go

```diff
@@ -672,10 +672,14 @@ func (nd *node) Commit(ctx context.Context, args *CommArgs) {
 		sendErr(err)
 		return
 	}
+
 	nd.tx.Close()
 	nd.tx = nil
 	nd.txmu.Unlock()
 
+	// Run the garbage collector to remove tagged Refs
+	nd.exch.Index().GC()
```

Comment:
Would be nice if you could add a check in `node_test.go` maybe in `TestCommit` or in a new test to make sure that 1) The node does implement the Index.GC correctly, 2) We do GC correctly when needed.